### PR TITLE
Added consistent spelling

### LINKED ISSUE #579: Use consistent spelling for Nox in documentation
Currently, docstrings, option help texts, and rst files use various spellings for Nox and the noxfile.py:

- nox
- Nox
- noxfile
- nox file
- Noxfile

We should make these usages consistent. For Nox itself, the case is already quite clear:

- Nox is spelled with a capital N, as a general rule.
- As an exception to this rule, nox is spelled in lowercase when referring to the executable. (Use sparingly.)

For the noxfile.py file, I'd propose *Noxfile* with capital N, just to get the bikeshed going. Any preferences/thoughts?

**Update:** Looking at some more of @theacodes' original commits here, I believe that "Nox" and "Noxfile" are indeed the preferred usage.

### PR DESCRIPTION
**Pull request by CN-M**
Closes #579

This PR aims to fix the inconsistent spelling of `Nox` and `Noxfile` all while leaving the `nox` executables and `noxfile.py` files unchanged.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -58,7 +58,7 @@
 master_doc = "index"
 
 # General information about the project.
-project = "Nox"
+project = "nox"
```
Comment (DiddiLeija) [inline docs/conf.py:61]: I consider this should keep as `Nox`, because it is a project?

File: CHANGELOG.md [hunk 2]
```diff
@@ -126,7 +126,7 @@ Special thanks to our contributors:
 - Support users specifying an undeclared parametrization of python via `--extra-python` (#361)
 - Support double-digit minor version in `python` keyword (#367)
 - Add `py.typed` to `manifest.in` (#360)
-- Update nox to latest supported python versions. (#362)
+- Update Nox to latest supported python versions. (#362)
```
Comment (DiddiLeija) [inline CHANGELOG.md:129]: I'm not sure about modifying the changelog, but maybe it's just me.

**Review by DiddiLeija — COMMENTED** [review]
I'm concerned by some things, and pointed them below. Anyway, probably it's just me, so if other maintainers disagree, I'm ok with these changes :+1:

**Comment by CN-M** [comment]
> I'm concerned by some things, and pointed them below. Anyway, probably it's just me, so if other maintainers disagree, I'm ok with these changes 👍

**Comment by CN-M** [comment]
Shit. Sorry. Trying to get used to this PR thing. Closed the PR by accident.
It's back open

**Comment by DiddiLeija** [comment]
I just edited your first post to link #579 to this PR, so it can be closed when merging this :)

**Review by cjolowicz — COMMENTED** [review]
@CN-M Thanks for working on this, looks great. Have you seen @DiddiLeija 's comment on the Sphinx config?

File: docs/conf.py (hunk 1 above)
Comment (DiddiLeija) [reply docs/conf.py:61]: If no one disagrees, this is my suggestion:

```suggestion
project = "Nox"
```

**Review by DiddiLeija — APPROVED** [review]
Now this sounds good to me!

File: docs/conf.py (hunk 1 above)
Comment (CN-M) [reply docs/conf.py]: @DiddiLeija, I applied your suggestion to the code. And I'm open to any other suggestions.

**Comment by CN-M** [comment]
Lovely to hear, @DiddiLeija 
It was really fun to work on this!

**Review by cjolowicz — APPROVED** [review]
Thank you!

**Pull request merged**